backend/src/chat/gateway/chat/chat.gateway.ts

```diff
@@ -393,18 +418,18 @@ export class ChatGateway implements OnGatewayConnection {
 			// who have not blocked the sender
 			// Speed here could probably be improved
 			const userIdsInChannel = this.channelIdToUserIds.get(channelId);
-			console.log('userIdsInChannel: ', userIdsInChannel);
 			if (userIdsInChannel) {
-				userIdsInChannel.forEach(async userId => {
-					if (!(await this.chatService.isUserBlocked(senderId, userId))) {
-						console.log('Unblocked message triggered for userId: ' + userId);
-						const socket = this.userIdToSocketMap.get(userId);
-						socket.emit('message', {
-							channelId: channelId,
-							senderId: senderId,
-							message: message,
-						});
-					} else {
+        for (const userId of userIdsInChannel) {
+          if (!(await this.chatService.isUserBlocked(senderId, userId))) {
+            console.log('UserId: ' + userId + 'received a message: ' + message);
+            const socket = this.userIdToSocketMap.get(userId);
+            socket.emit('message', {
```

Comment:
Add protection here please: `socket?.emit`

Reply:
I'll handle this on the other PR so Eduardo can have an updated version from dev